crates/bevy_render/src/render_graph/command.rs

```diff
@@ -147,7 +147,13 @@ impl CommandQueue {
         self.queue.lock().clear();
     }
 
-    pub fn execute(&mut self, render_context: &mut dyn RenderContext) {
+    pub fn take_commands(&mut self, command_queue: &mut CommandQueue) {
+        self.queue
+            .lock()
+            .extend(command_queue.queue.lock().drain(..))
+    }
```

Comment:
If you intend for the `CommandQueue` to not be shared at this point as it seems from the function signature, you could use `Mutex::get_mut(Arc::get_mut(&mut command_queue.queue).unwrap()).unwrap()` to avoid a lock and potential deadlock on something like `let mut queue2 = queue.clone(); queue.take_commands(&mut queue2)`.

Reply:
I do like the idea of removing locks, but doesn't create a situation where it could panic it runtime if `take_commands` is called on two commands instances at the same time? Maybe using something like crossbeam_channel might be the better solution for this.

I just realized `take_commands` isn't even used anywhere right now, so I'm inclined to just remove it.

Reply:
I was thinking abecause you take `&mut self` instead of `&self`, did you intend for the inner queue to not be shared at that point? Otherwise `&self` would make more sense.

Reply:
It would still be "shared" in the sense that there would be two CommandQueues with the same Arc. But the motivating use-case for the method did not have concurrent mutex accesses.